app.test.js

```diff
@@ -44,11 +44,70 @@ describe('Server', () => {
 
       })
   });
+
   describe('GET api/v1/projects/:id', () => {
     it('should return a 200 status and a single project if the project exists', async () => {
 
     })
-  })
+  }); 
 
+  describe('GET api/v1/palettes/:id', () => {
+    it('should return a 200 status and a single palette', async () => { 
+        const palette = await database('palettes').first();
+        const { id } = palette;
+        const onePalette = JSON.parse(JSON.stringify(palette));
+        const response = await request(app).get(`/api/v1/palettes/${id}`);
+        const result = response.body[0];
 
+        expect(response.status).toBe(200);
+        expect(result).toEqual(onePalette);
+    })
+
+    it('should return a 404 and the message "Pallete not found"', async () => {
+      const invalidId = -555;
+      const response = await request(app).get(`/api/v1/palettes/${invalidId}`);
+     
+      expect(response.status).toBe(404);
+      expect(response.body.error).toEqual('Pallete not found');
+    });
+  }); 
+ 
+  describe('POST /api/v1/palettes', () => {
+      it('should post a new palette to the db', async () => {
+        
+        const newPalette = { name: 'Luna Llena', color1 : '#F7EDB7', color2: '#00A8CF', color3: '#B3C0F7', color4: '#DFE9FD', color5: '#DFE9FD'};
+        const response = await request(app).post('/api/v1/palettes').send(newPalette);
+        const palettes = await database('palettes').where('id', response.body.id[0]);
+        const palette = palettes[0];
+  
+        expect(response.status).toBe(201);
+        expect(palette.name).toEqual(newPalette.name);
+      });
+  });
+  
+  describe('PATCH /api/v1/palettes/:id', () => {
+    it('should patch a palette to update info', async () => {
+          const mockName = { name: 'Mar de Flores'};
+          const palette = await database('palettes').first();
```

Comment:
Hey!  So I think we need an extra line of code in here so that this isn't a false positive - there's a chance that this would pass anyway if the name was already 'Mar de Flores'.  I think it would help to stick an 'expected' extra line in here on line 92 confirming that the name is NOT Mar de Flores before the patch happens.

Reply:
Will deffo fix =)

Reply:
Thanks!  Looks great!